[EDU-3169] feat: add automate secure - terraform (#456)
aziontech/docs: src/content/docs/en/pages/automate-terraform.mdx

---
title: Automate security with Azion Terraform provider

> description: Azion Terraform Provider is an open source project, registered in HashiCorp. Through the Azion SDK for Go, the provider communicates with the Azion APIs, so you can manage your infrastructure hosted on the the Azion platform, locally, as code.

Reviewer: ```suggestion
description: Azion Terraform Provider is an open source project, registered in HashiCorp. Through the Azion SDK for Go, the provider communicates with the Azion APIs, so you can manage your infrastructure hosted on the Azion's platform, locally, as code.
```

meta_tags: edge, security
namespace: documentation_secure_automate_terraform

> menu_namespace: devtoolsMenu

Reviewer 2: ```suggestion
menu_namespace: secureMenu
```

permalink: /documentation/products/secure/automate/terraform/
---

[Azion Terraform Provider](https://registry.terraform.io/providers/aziontech/azion/latest/docs) is a robust tool that integrates seamlessly with Terraform, the industry-leading Infrastructure as Code (IaC) platform.

## Security with Terraform

> In modern web, security is key. Terraform addresses this concern by codifying infrastructure. This approach allows security configurations to be defined, versioned, and audited in tandem with application code. This ensures uniformity across different environments 
> and significantly reduces the risk of misconfigurations.

Reviewer 2: ```suggestion
In modern web, security is key. Terraform addresses this concern by codifying infrastructure. This approach allows security configurations to be defined, versioned, and audited in tandem with application code, ensuring uniformity across different environments and significantly reducing the risk of misconfigurations.
```

One of the primary benefits of Terraform is its automation capabilities. It simplifies repetitive tasks, thereby reducing the likelihood of human error and accelerating deployment processes. By codifying infrastructure, Terraform enables the use of reusable modules and templates. This not only boosts efficiency but also ensures the consistent application of security measures throughout the infrastructure lifecycle.

## Azion Terraform Provider and security

| Scope | Resource | Data source |
| --- | --- | --- |
|  Edge Firewall Main Settings | [Managing main settings](https://registry.terraform.io/providers/aziontech/azion/latest/docs/resources/edge_firewall_main_setting) | [Consulting main settings](https://registry.terraform.io/providers/aziontech/azion/latest/docs/data-sources/edge_firewall_main_settings) |
|  Edge Firewall Edge Functions Instances | [Managing edge functions instances](https://registry.terraform.io/providers/aziontech/azion/latest/docs/resources/edge_firewall_edge_functions_instance) | [Consulting edge functions instances](https://registry.terraform.io/providers/aziontech/azion/latest/docs/data-sources/edge_firewall_edge_function_instances) |

> |  Network list| [Managing network lists](https://registry.terraform.io/providers/aziontech/azion/latest/docs/resources/network_list) | [Consulting network lists](https://registry.terraform.io/providers/aziontech/azion/latest/docs/data-sources/network_lists) |

Reviewer 2: ```suggestion
|  Network list | [Managing network lists](https://registry.terraform.io/providers/aziontech/azion/latest/docs/resources/network_list) | [Consulting network lists](https://registry.terraform.io/providers/aziontech/azion/latest/docs/data-sources/network_lists) |
```
